TypeScript.py

```diff
@@ -397,8 +402,9 @@ def setFilePrefs(view):
     settings = view.settings()
     settings.set('use_tab_stops', False)
     settings.set('translate_tabs_to_spaces', True)
-    settings.add_on_change('tab_size',lambda: tab_size_changed(view))
-    settings.add_on_change('indent_size',lambda: tab_size_changed(view))
+    if (settings and (settings.add_on_change)):
+        settings.add_on_change('tab_size',lambda: tab_size_changed(view))
+        settings.add_on_change('indent_size',lambda: tab_size_changed(view))
```

Comment:
These are un-related but I think subsumed by later changes.  They are for updating format configuration when it changes per file.